app/controllers/topics_controller.rb

```diff
@@ -43,6 +43,13 @@ def id_for_slug
     render json: { slug: topic.slug, topic_id: topic.id, url: topic.url }
   end
 
+  def show_by_external_id
+    topic = Topic.find_by(external_id: params[:external_id])
+    guardian.ensure_can_see!(topic)
+    raise Discourse::NotFound unless topic
```

Comment:
```suggestion
    topic = Topic.find_by(external_id: params[:external_id])
    raise Discourse::NotFound unless topic
    guardian.ensure_can_see!(topic)
```

I think the order is wrong here. We need to check whether the topic is present first before passing it to the guardian.